### PR TITLE
MEN-4584: store/v2, single-database multitenancy model

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: store/v2/utils.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		}
+	case bson.D:
+		res = append(res, v...)
+	default:
+		if data, err := bson.Marshal(v); err == nil {
+			bsonData := bson.D{}
+			if err = bson.Unmarshal(data, &bsonData); err == nil {
+				res = append(res, bsonData...)
+			}
+		}
```
Comment (alfrunes) [inline store/v2/utils.go:56]: There is a function [mongo/doc.DocumentFromStruct](https://github.com/mendersoftware/go-lib-micro/blob/27b9a510e50adda7e6bc08e6300ba86eadd0d4cd/mongo/doc/bson.go#L29) that does this using reflection.

File: store/v2/utils.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	return res
+}
+
+// ArrayWithTenantID adds the tenant_id field to an array of bson documents
+// using the value extracted from the identity of the context
+func ArrayWithTenantID(ctx context.Context, doc bson.A) bson.A {
```
Comment (alfrunes) [inline store/v2/utils.go:64]: I guess this could also be implemented as a case in `WithTenantID`?

**Review by alfrunes — APPROVED** [review]
LGTM
(I'm just thinking out loud).

**Review by mchalski — APPROVED** [review]

File: store/v2/utils.go (hunk 2 above)
Comment (tranchitella) [reply store/v2/utils.go:64]: It was tricky because they have different return types. How could I do it?

File: store/v2/utils.go (hunk 1 above)
Comment (tranchitella) [reply store/v2/utils.go]: Thanks, I'm reusing it

**Pull request merged**